### PR TITLE
Add self-documentation of TLS resources

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

**Comment by soltysh** [comment]
@sttts are you planning on finishing this or closing?

**Comment by sttts** [comment]
> @sttts are you planning on finishing this or closing?

I still think it is the right approach, but didn't find time for finishing it.

File: TLS.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+<details><summary>openshift-kube-apiserver-operator/csr-controller-signer-ca</summary><blockquote>
+
+## openshift-kube-apiserver-operator/csr-controller-signer-ca
+
+ConfigMap provided by the ???.
+</blockquote></summary></details>
+
+## openshift-config-managed/kube-controller-manager-client-cert-key
+
+Rotated key, validity 720h0m0s, refreshed every 360h0m0s.
```
Comment (mfojtik) [inline TLS.md:21]: removing `0s` makes this little bit more readable

File: TLS.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+## openshift-kube-apiserver-operator/csr-signer-ca
+
+ConfigMap
+
+</blockquote></summary></details>
+<details><summary>openshift-kube-apiserver-operator/csr-controller-signer-ca</summary><blockquote>
+
+## openshift-kube-apiserver-operator/csr-controller-signer-ca
+
+ConfigMap provided by the ???.
```
Comment (mfojtik) [inline TLS.md:16]: TODO? we should be able to tell the provider

File: TLS.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
+# TLS Documentation
+
+## openshift-kube-apiserver-operator/csr-controller-ca
```
Comment (mfojtik) [inline TLS.md:3]: i think each cert deserves one or two sentence description?

File: TLS.md [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+CA bundle ConfigMap containing these CAs:<details><summary>openshift-kube-apiserver-operator/csr-signer-ca</summary><blockquote>
+
+## openshift-kube-apiserver-operator/csr-signer-ca
+
+ConfigMap
+
+</blockquote></summary></details>
+<details><summary>openshift-kube-apiserver-operator/csr-controller-signer-ca</summary><blockquote>
```
Comment (mfojtik) [inline TLS.md:12]: is this the namespace for this configmap? what does this CA do?

File: tls/rotation.go [hunk 5]
```diff
@@ -0,0 +1,167 @@
+package tls
+
+import (
+	"time"
+)
+
+var (
+	kubeControlPlaneSignerConfig = SignerConfig{
+		Namespace: "openshift-kube-apiserver-operator", Name: "kube-control-plane-signer",
+		Validity: 60 * 24 * time.Hour, // this comes from the installer TODO: what does this mean?
```
Comment (mfojtik) [inline tls/rotation.go:10]: i wonder if we can use something like https://github.com/xhit/go-str2duration and write down "60d" here :-)

**Pull request closed without merging**